### ISSUE TITLE
HomeController's Index View improperly redirects to the correct URL

### ISSUE THREAD

**Issue by tessierp** [IdentityServer/IdentityServer4.Quickstart.UI#34]
We are using the IdentityServer 4 behind a Load Balancer and in Docker and using Docker Flow Proxy to terminate the SSL. We followed the instructions we found here to solve the issues assosicated with this configuration :  

https://leastprivilege.com/2017/10/09/new-in-identityserver4-v2-simplified-configuration-behind-load-balancers-or-reverse-proxies/comment-page-1/#comment-92277

Works perfectly except in one case. In the views that are graciously offered by your team, we have in the HomeController’s the following :

"Click here to manage your stored grants", where in the "HERE" in index.cshtm of the HomeController is generated with ~/grants. This causes a generation of the wrong URL, http://xxx.com/account/login?returnUrl=%2Fgrants, instead of https://xxx.com/account/login?returnUrl=%2Fgrants

I assume this is because in the configuration of Kestrel it knows it is set up for HTTP and thus generates the URL in http. Is there an option I can set where the IdentityServer overrides this and sets it to HTTPS?

**Comment by brockallen** [comment]
Feel free to send a PR with the correct redirect path string to work behind a load balancer.

**Comment by brockallen** [comment]
I started looking at this, and I think it's a larger problem than we thought. In short, by setting `PublicOrigin` that only really informs IdentityServer's internals about the origin, but not MVC or any other custom code, thus for now that really still falls back to you as the developer of the host to configure the load balancer infrastructure properly.

We can look at trying to emulate the ForwardedHeadersMiddleware from Microsoft based on this setting. I've opened a new issue here as a feature request to track future consideration: https://github.com/IdentityServer/IdentityServer4/issues/1834
